examples/imagenet_cnn/driver.py

```diff
@@ -0,0 +1,477 @@
+  return sum([ray.get(shape_ref)[0] for shape_ref in shape_refs])
+
+@ray.remote([List], [np.ndarray])
+def compute_mean_image(batches):
+  """Computes the mean of images in batches.
+  
+  Args:
+    batches (List): Collection of batches of images and labels.
+  
+  Returns:
+    ndarray: The mean image
+  """
+
+  if len(batches) == 0:
+    raise Exception("No images were passed into `compute_mean_image`.")
+  sum_image_refs = [ra.sum(batch[0], axis=0) for batch in batches]
+  sum_images = [ray.get(ref) for ref in sum_image_refs]
+  n_images = num_images(batches)
+  return np.sum(sum_images, axis=0).astype("float64") / ray.get(n_images)
+
+@ray.remote([np.ndarray, np.ndarray, np.ndarray, np.ndarray], [np.ndarray, np.ndarray, np.ndarray, np.ndarray])
+def shuffle_arrays(images1, labels1, images2, labels2):
+  """Shuffles the data of two batches.
+  
+  Args:
+    images1 (ndarray): Images of first batch
+    labels1 (ndarray): Labels of first batch
+    images2 (ndarray): Images of second batch
+    labels2 (ndarray): Labels of second batch
+  
+  Returns:
+    ndarray: Shuffled images for first batch
+    ndarray: Shuffled labels for first batch
+    ndarray: Shuffled images for second batch
+    ndarray: Shuffled labels for second batch
+  """
+  to_shuffle = zip(images1, labels1) + zip(images2, labels2)
+  np.random.shuffle(to_shuffle)
+  length = len(images1)
+  first_batch = zip(*to_shuffle[0:length])
+  first_array = map(np.asarray, first_batch)
+  second_batch = zip(*to_shuffle[length:len(to_shuffle)])
+  second_array = map(np.asarray, second_batch)
+  return first_array[0], first_array[1], second_array[0], second_array[1]
+
+@ray.remote([Tuple, Tuple], [Tuple])
+def shuffles_tuples(first_batch, second_batch):
+  """Calls shufflearrays in order to properly serialize the data.
+  Args:
+    first_batch (tuple): First batch to be shuffled.
+    second_batch (tuple): Second batch to be shuffled
+  
+  Returns:
+    Tuple: Two batches of images and labels.
+  """
+  if (first_batch == None) or (second_batch == None):
+    return (first_batch, second_batch)
+  shuffled_batches = shuffle_arrays(first_batch[0], first_batch[1], second_batch[0], second_batch[1])
+  return ((shuffled_batches[0], shuffled_batches[1]), (shuffled_batches[2], shuffled_batches[3]))
+
+@ray.remote([list, dict], [np.ndarray])
+def convert(img_list, image_pairs):
+  """Converts filename strings to integer labels.
+  
+  Args:
+    imglist (List): Filenames
+    imagepairs (dict): Lookup table for filenames
+  Returns:
+    ndarray: Integer labels
+  """
+  return np.asarray(map(lambda img_name:int(image_pairs[img_name]), img_list))
+
+def one_hot(x):
+  """Converts integer labels to one hot vectors.
+  
+  Args:
+    x (int): Index to be set to one
+  Returns:
+    ndarray: One hot vector.
+  """
+  zero = np.zeros([1000])
+  zero[x] = 1.0
+  return zero
+
+def crop_image(img):
+  """Crops an input image to prove more training for the network.
```

Comment:
How about a little more description here:

"Generate a random crop of an image. This is used to generate many slightly different images from each training example to improve the network's capability to generalize."